### PR TITLE
QG-8 - User api

### PR DESCRIPTION
**Pull request by nvminhtam**

### CODE REVIEW

File: src/services/user.service.ts [hunk 1]
```diff
@@ -3,6 +3,10 @@ import { v4 as uuidv4 } from 'uuid';
 import bcrypt from 'bcrypt';
 import { models } from '../models';
 
+import path from 'path';
+import nodemailer from 'nodemailer';
+const hbs = require('nodemailer-express-handlebars');
```
Comment (huydoan1908) [inline src/services/user.service.ts]: should use import instead of require

File: src/controllers/auth.controller.ts [hunk 2]
```diff
@@ -3,8 +3,16 @@ import bcrypt from 'bcrypt';
...
 import { UserRegisterInfo } from '../interfaces/user.interface';
-import { findUser, createUser, checkEmail } from '../services/user.service';
+import {
+  findUser,
+  createUser,
+  checkEmail,
+  sendActivationEmail,
+  updateAccountActivation,
+} from '../services/user.service';
+import console from 'console';
```
Comment (huydoan1908) [inline src/controllers/auth.controller.ts]: What is this use for ?

File: src/controllers/auth.controller.ts [hunk 3]
```diff
@@ -49,6 +71,13 @@ const login = async (req: Request, res: Response) => {
         message: 'Username or password is incorrect.',
       });
     }
+    if (user.active == false) {
```
Comment (huy27201) [inline src/controllers/auth.controller.ts]: should `===` instead of `==`

File: src/controllers/auth.controller.ts [hunk 4]
```diff
@@ -124,19 +154,126 @@ const loginWithGoogle = async (req: Request, res: Response) => {
...
+          success: false,
+          message: 'Token validate fail',
+        });
+      }
+      const { username, email } = decodedToken;
+      var user = await findUser({ username, email });
+      if (!user) {
+        return res.status(401).json({
+          success: false,
+          message: 'This account is not exists!',
```
Comment (huy27201) [inline src/controllers/auth.controller.ts]: should be `does not exist`

**Review by huy27201 — APPROVED** [review]

**Pull request merged**